frontend/src/components/ui/locale-picker.ts

```diff
@@ -1,28 +1,31 @@
 import { localized } from "@lit/localize";
-import { html, LitElement } from "lit";
+import type { SlSelectEvent } from "@shoelace-style/shoelace";
+import { html } from "lit";
 import { customElement, state } from "lit/decorators.js";
 
-import { allLocales } from "@/__generated__/locale-codes";
-import { getLocale, setLocaleFromUrl } from "@/utils/localization";
+import { sourceLocale } from "@/__generated__/locale-codes";
+import { BtrixElement } from "@/classes/BtrixElement";
+import { allLocales, type LocaleCodeEnum } from "@/types/localization";
+import { getLocale, setLocale } from "@/utils/localization";
+import { AppStateService } from "@/utils/state";
 
-type LocaleCode = (typeof allLocales)[number];
 type LocaleNames = {
-  [L in LocaleCode]: string;
+  [L in LocaleCodeEnum]: string;
 };
 
 @localized()
 @customElement("btrix-locale-picker")
-export class LocalePicker extends LitElement {
+export class LocalePicker extends BtrixElement {
   @state()
   private localeNames: LocaleNames | undefined = {} as LocaleNames;
 
-  private readonly setLocaleName = (locale: LocaleCode) => {
+  private readonly setLocaleName = (locale: LocaleCodeEnum) => {
```

Comment:
I'm wondering if we want to capitalize the name that comes out of this for consistency?

Reply:
Capitalization rules are different for each language, so we should leave this to the translator. Spanish specifically doesn't capitalize languages, but I'm guessing it could follow a different rule if it's the sole word--I'll add more descriptive text and and a screenshot for translators?

Reply:
cc @DaleLore ^